flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/filesystem/DeserializationSchemaAdapter.java

```diff
@@ -73,31 +73,32 @@
 
     public DeserializationSchemaAdapter(
             DeserializationSchema<RowData> deserializationSchema,
-            TableSchema schema,
+            List<Column> physicalColumns,
```

Comment:
We should just pass the (physical) `DataType` here and keep that. It contains both names and types.